### PR TITLE
Raised PR For Different Formatting Cards Nodejs

### PR DESCRIPTION
**Pull request by Mohammed-MSFT**

### CODE REVIEW

File: samples/bot-formatting-cards/nodejs/appPackage/manifest.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "privacyUrl": "https://www.microsoft.com/privacy",
+    "termsOfUseUrl": "https://www.microsoft.com/termsofuse"
+  },
+  "name": {
+    "short": "Different Formatting Cards",
+    "full": "Different Formatting Cards"
+  },
+  "description": {
+    "short": "Different Formatting Cards",
+    "full": "This sample demonstrates the use of different formatting on Cards."
```
Comment (Pawan-MSFT) [inline samples/bot-formatting-cards/nodejs/appPackage/manifest.json:19]: 1. Can we add "Adaptive" before cards? like "This sample demonstrates the use of different formatting on adaptive Cards" and "Different Formatting Adaptive Cards"
2. "This sample demonstrates the use of different formatting on Cards." , "C" should small

File: samples/bot-formatting-cards/nodejs/assets/sample.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
+[
+  {
+    "name": "officedev-microsoft-teams-samples-bot-formatting-cards-nodejs",
+    "source": "officeDev",
+    "title": "Different Formatting On Cards",
+    "shortDescription": "Demonstrating the feature where user can send different formatting on cards.",
+    "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/bot-formatting-cards/nodejs",
+    "longDescription": [
+      "Demonstrating the feature where user can send different formatting on cards using bot."
```
Comment (Pawan-MSFT) [inline samples/bot-formatting-cards/nodejs/assets/sample.json:9]: same for here also "Demonstrating the feature where user can send different formatting on adaptive cards using bot."

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+    /**
+    * Sends Mention Support Card
+    */
+    sendMentionSupportCard() {
+        return CardFactory.adaptiveCard(MentionSupport);
+    }
+
+    /**
+    * Sends Sample AdaptiveCard with full width
```
Comment (Pawan-MSFT) [inline samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js:104]: Please update "Sends Sample Adaptive Card With Full Width"

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+            // By calling next() you ensure that the next BotHandler is run.
+            await next();
+        });
+
+        this.onMessage(async (context, next) => {
+            const text = context.activity.text;
+
+            // Create an array with the valid card options.
+            const adaptiveFormatCards = ['MentionSupport', 'InfoMasking', 'SampleCard', 'StageViewImages', 'OverflowMenu', 'HTMLConnector', 'CardWithEmoji'];
```
Comment (Pawan-MSFT) [inline samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js:30]: Change SampleCard to 'FullWidthCard".

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+                value: 'MentionSupport'
+            },
+            {
+                type: ActionTypes.ImBack,
+                title: 'InfoMasking',
+                value: 'InfoMasking'
+            },
+            {
+                type: ActionTypes.ImBack,
+                title: 'SampleCard',
```
Comment (Pawan-MSFT) [inline samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js:163]: Change SampleCard to 'FullWidthCard".

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+                value: 'SampleCard'
```
Comment (Pawan-MSFT) [inline samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js:164]: Change SampleCard to 'FullWidthCard".

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+                switch (text) {
+                    case "MentionSupport":
+                        await context.sendActivity({ attachments: [this.sendMentionSupportCard()] });
+                        break;
+
+                    case "InfoMasking":
+                        await context.sendActivity({ attachments: [this.sendInfoMasking()] });
+                        break;
+
+                    case "SampleCard":
```
Comment (Pawan-MSFT) [inline samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js:44]: Change SampleCard to 'FullWidthCard".

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+                        await context.sendActivity({ attachments: [this.sendSampleAdaptiveCard()] });
```
Comment (Pawan-MSFT) [inline samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js:45]: Change SendSampleAdaptiveCard to 'SendfullWidthCard".

**Review by Pawan-MSFT — COMMENTED** [review]
few comments please fix

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
+---
+page_type: sample
+description: This sample demonstrates the use of different formatting on cards
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:3]: This sample shows how to use different formatting on adaptive card.

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+- nodejs
+extensions:
+ contentType: samples
+ createdDate: "01/28/2023 05:00:17 PM"
+urlFragment: officedev-microsoft-teams-samples-bot-formatting-cards-nodejs
+
+---
+# Different Formatting Cards
+
+This sample shows the feature where user can send different formatting on cards using bot.
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:18]: This sample shows the feature where user can use different formatting on adaptive cards using bot.

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+
+## Interaction with app
+
+![Types Of Cards](Images/DifferentFormattingCards.gif)
+
+## Prerequisites
+
+-  Microsoft Teams is installed and you have an account (not a guest account).
+-  To test locally, [NodeJS](https://nodejs.org/en/download/) must be installed on your development machine (version 18x  or higher).
+-  [ngrok](https://ngrok.com/) or equivalent tunneling solution.
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:28]: Link .zip folder here

File: samples/bot-formatting-cards/nodejs/appPackage/manifest.json [hunk 12]
```diff
@@ -0,0 +1,38 @@
+﻿{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.14/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.14",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/appPackage/manifest.json:3]: Can we use latest version ?

File: samples/bot-formatting-cards/nodejs/assets/sample.json (hunk 2 above)
Comment (ChetanSharma-msft) [reply samples/bot-formatting-cards/nodejs/assets/sample.json:9]: Refer this PR: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/689/files#diff-824fbd3defb344889effcfecba9c5a1ae303b488e565c47c32494597545bba6b

File: samples/bot-formatting-cards/nodejs/appPackage/manifest.json (hunk 1 above)
Comment (ChetanSharma-msft) [reply samples/bot-formatting-cards/nodejs/appPackage/manifest.json:19]: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/689/files#diff-824fbd3defb344889effcfecba9c5a1ae303b488e565c47c32494597545bba6b

File: samples/bot-formatting-cards/nodejs/resources/adaptiveCard.json [hunk 13]
```diff
@@ -0,0 +1,29 @@
+﻿{
+  "$schema": "https://adaptivecards.io/schemas/adaptive-card.json",
+  "type": "AdaptiveCard",
+  "version": "1.0",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/adaptiveCard.json:4]: Refer this PR: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/689/files#diff-824fbd3defb344889effcfecba9c5a1ae303b488e565c47c32494597545bba6b

File: samples/bot-formatting-cards/nodejs/resources/adaptiveCardWithEmoji.json [hunk 14]
```diff
@@ -0,0 +1,24 @@
+﻿{
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "type": "AdaptiveCard",
+  "version": "1.0",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/adaptiveCardWithEmoji.json:4]: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/689/files#diff-824fbd3defb344889effcfecba9c5a1ae303b488e565c47c32494597545bba6b

File: samples/bot-formatting-cards/nodejs/resources/informationMasking.json [hunk 15]
```diff
@@ -0,0 +1,32 @@
+﻿{
+  "$schema": "https://adaptivecards.io/schemas/adaptive-card.json",
+  "type": "AdaptiveCard",
+  "version": "1.0",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/informationMasking.json:4]: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/689/files#diff-824fbd3defb344889effcfecba9c5a1ae303b488e565c47c32494597545bba6b

File: samples/bot-formatting-cards/nodejs/resources/mentionSupport.json [hunk 16]
```diff
@@ -0,0 +1,37 @@
+﻿{
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "version": "1.0",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/mentionSupport.json:3]: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/689/files#diff-824fbd3defb344889effcfecba9c5a1ae303b488e565c47c32494597545bba6b

File: samples/bot-formatting-cards/nodejs/resources/sampleAdaptiveWithFullWidth.json [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+                },
+                {
+                    "title": "Blue",
+                    "value": "3"
+                }
+            ]
+        }
+    ],
+    "$schema": "https://adaptivecards.io/schemas/adaptive-card.json",
+    "version": "1.0",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/sampleAdaptiveWithFullWidth.json:56]: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/689/files#diff-824fbd3defb344889effcfecba9c5a1ae303b488e565c47c32494597545bba6b

File: samples/bot-formatting-cards/nodejs/resources/mentionSupport.json [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+      "text": "Hi <at>Adele UPN</at>, <at>Adele Azure AD</at>"
+    }
+  ],
+  "msteams": {
+    "entities": [
+      {
+        "type": "mention",
+        "text": "<at>Adele UPN</at>",
+        "mentioned": {
+          "id": "AdeleV@contoso.onmicrosoft.com",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/resources/mentionSupport.json:23]: https://github.com/OfficeDev/Microsoft-Teams-Samples/pull/689/files

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments

**Review by Wajeed-msft — APPROVED** [review]

File: samples/bot-formatting-cards/nodejs/appPackage/manifest.json (hunk 1 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/appPackage/manifest.json]: Fixed

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js (hunk 3 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js]: Fixed

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js (hunk 5 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js]: Fixed

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js (hunk 7 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 9 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/sampleAdaptiveWithFullWidth.json (hunk 17 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/sampleAdaptiveWithFullWidth.json]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/mentionSupport.json (hunk 16 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/mentionSupport.json]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/informationMasking.json (hunk 15 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/informationMasking.json]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/adaptiveCardWithEmoji.json (hunk 14 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/adaptiveCardWithEmoji.json]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/adaptiveCard.json (hunk 13 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/adaptiveCard.json]: Fixed

File: samples/bot-formatting-cards/nodejs/appPackage/manifest.json (hunk 12 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/appPackage/manifest.json]: Fixed

File: samples/bot-formatting-cards/nodejs/resources/mentionSupport.json (hunk 18 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/resources/mentionSupport.json]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 11 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md:33]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 10 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md]: Fixed

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js (hunk 8 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js]: Fixed

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js (hunk 6 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js]: Fixed

File: samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js (hunk 4 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/bots/botFormattingCards.js]: Fixed

File: samples/bot-formatting-cards/nodejs/assets/sample.json (hunk 2 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/assets/sample.json]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 19]
```diff
@@ -0,0 +1,169 @@
...
+- office-365
+languages:
+- nodejs
+extensions:
+ contentType: samples
+ createdDate: "01/28/2023 05:00:17 PM"
+urlFragment: officedev-microsoft-teams-samples-bot-formatting-cards-nodejs
+
+---
+# Different Formatting Cards
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:16]: ## Different Formatting Cards

File: samples/bot-formatting-cards/nodejs/Readme.md [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+> Note these instructions are for running the sample on your local machine, the tunnelling solution is required because the Teams service needs to call into the bot.
+
+1. Register a new application in the [Azure Active Directory – App Registrations](https://go.microsoft.com/fwlink/?linkid=2083908) portal.
+
+2. Setup for Bot
+- In Azure portal, create a [Azure Bot resource](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0&tabs=csharp%2Caadv2).
+- Ensure that you've [enabled the Teams Channel](https://docs.microsoft.com/en-us/azure/bot-service/channel-connect-teams?view=azure-bot-service-4.0)
+- While registering the bot, use `https://<your_ngrok_url>/api/messages` as the messaging endpoint.
+
+	**NOTE:** When you create your bot you will create an App ID and App password - make sure you keep these for later.
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/Readme.md:46]: Alignment looks wrong

File: samples/bot-formatting-cards/nodejs/appPackage/manifest.json [hunk 21]
```diff
@@ -0,0 +1,38 @@
+﻿{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.14/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.15",
```
Comment (ChetanSharma-msft) [inline samples/bot-formatting-cards/nodejs/appPackage/manifest.json:3]: Wrong, URL is referring 1.14

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added comments

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 19 above)
Comment (ChetanSharma-msft) [reply samples/bot-formatting-cards/nodejs/Readme.md:16]: ##

Comment (ChetanSharma-msft) [reply samples/bot-formatting-cards/nodejs/Readme.md:16]: `## Different Formatting Cards`

**Comment by ChetanSharma-msft** [comment]
Kindly execute build pipeline

File: samples/bot-formatting-cards/nodejs/appPackage/manifest.json (hunk 21 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/appPackage/manifest.json:3]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 19 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md]: Fixed

File: samples/bot-formatting-cards/nodejs/Readme.md (hunk 20 above)
Comment (Pawan-MSFT) [reply samples/bot-formatting-cards/nodejs/Readme.md]: Fixed

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Changes looks fine, approving it.

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Comment by ChetanSharma-msft** [comment]
@Pawan-MSFT - Please run build pipeline before pushing it.

**Comment by Pawan-MSFT** [comment]
Executed Build Pipeline :
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/4053814286
Result:
image

**Pull request merged**